unity/Assets/Scripts/Logic/TileStack.cs

```diff
@@ -43,7 +43,14 @@ public void PushTiles(List<ITile> tiles)
         // Pushes a new Tile
         public void Push(ITile newTile)
         {
-            stack.Add(newTile);
+            if(stack.Count == 0)
```

Comment:
Does it need the differentiation here?

Reply:
Yes. It's not possible to insert a tile to position 0 if the list is empty

Reply:
In that case, does the if-statement ever result in `true` (which in terms would just add another tile since `stack.Add(newTile);` gets called on line 48)?